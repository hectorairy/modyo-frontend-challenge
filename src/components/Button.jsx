import PropTypes from "prop-types";

import "../styles/Button.scss";

export const Button = ({ children, handleClick }) => {
  return (
    <button className="custom-button" onClick={handleClick}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  handleClick: PropTypes.func.isRequired,
};
