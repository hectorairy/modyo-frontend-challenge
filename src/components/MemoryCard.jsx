import PropTypes from "prop-types";

export const MemoryCard = ({ card, id, selectCard }) => {
  const { url_image, stat } = card;
  const classCard = stat ? `memory-game__card--active ${stat}` : "";

  const handleClick = () => {
    if (stat.includes("correct")) return false;
    selectCard(id);
  };

  return (
    <div className={`memory-game__card ${classCard}`} onClick={handleClick}>
      <img src={url_image} alt="" />
    </div>
  );
};

MemoryCard.propTypes = {
  card: PropTypes.object.isRequired,
  id: PropTypes.number.isRequired,
  selectCard: PropTypes.func.isRequired,
};
