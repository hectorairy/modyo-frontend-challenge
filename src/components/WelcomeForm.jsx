import { useState } from "react";
import PropTypes from "prop-types";

import "../styles/WelcomeForm.scss";

export const WelcomeForm = ({ setUser, username, setUsername }) => {
  const [error, setError] = useState(false);
  const handleSubmit = () => {
    setError(false);
    if (!username) {
      setError(true);
      return false;
    }
    localStorage.setItem("username", username);
    setUser(true);
  };
  return (
    <div className="welcome-form__container">
      <h2 className="welcome-form__title">¡Bienvenido a MemoryGame!</h2>
      <p className="welcome-form__subtitle">
        Nos encanta tenerte aquí, por favor dinos tu nombre para poder
        conocernos mejor. 😁
      </p>

      <input
        className="welcome-form__input"
        type="text"
        placeholder="Ej. Hector Jimenez"
        onChange={(e) => setUsername(e.target.value)}
        value={username}
      />

      {error && (
        <p className="welcome-form__error">
          Debes decirnos tu nombre antes de continuar 😢
        </p>
      )}

      <button className="welcome-form__button" onClick={handleSubmit}>
        ¡Empecemos a jugar!
      </button>
    </div>
  );
};

WelcomeForm.propTypes = {
  setUser: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired,
  setUsername: PropTypes.func.isRequired,
};
