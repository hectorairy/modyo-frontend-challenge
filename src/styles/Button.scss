.custom-button {
  @apply tw-font-bold;
  @apply tw-text-center;
  @apply tw-uppercase;
  @apply tw-transition-all;
  @apply tw-text-xs;
  @apply tw-py-3;
  @apply tw-px-6;
  @apply tw-rounded-lg;
  @apply tw-border;
  @apply tw-border-blue-500;
  @apply tw-text-blue-500;
  @apply hover:tw-opacity-75;
  @apply focus:tw-ring;
  @apply focus:tw-ring-blue-200;
  @apply active:tw-opacity-[0.85];
}
