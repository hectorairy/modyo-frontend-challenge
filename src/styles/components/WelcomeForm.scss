.welcome-form {
  &__container {
    @apply tw-w-72;
    @apply md:tw-w-96;
    @apply tw-bg-white;
    @apply tw-rounded;
    @apply tw-px-4;
    @apply tw-py-8;
    @apply tw-text-center;
  }

  &__title {
    @apply tw-font-bold;
    @apply tw-text-xl;
    @apply tw-mb-2;
  }

  &__subtitle {
    @apply tw-text-justify;
    @apply tw-mb-6;
  }

  &__input {
    @apply tw-appearance-none;
    @apply tw-border;
    @apply tw-border-blue-100;
    @apply tw-rounded;
    @apply tw-w-full;
    @apply tw-py-2;
    @apply tw-px-4;
    @apply tw-mb-4;
    @apply focus:tw-outline-none;
    @apply focus:tw-bg-white;
    @apply focus:tw-border-blue-500;
  }

  &__error {
    @apply tw-text-red-500;
    @apply tw-mb-4;
  }

  &__button {
    @apply tw-font-bold;
    @apply tw-text-center;
    @apply tw-uppercase;
    @apply tw-transition-all;
    @apply tw-text-xs;
    @apply tw-py-3;
    @apply tw-px-6;
    @apply tw-rounded-lg;
    @apply tw-border;
    @apply tw-border-blue-500;
    @apply tw-text-blue-500;
    @apply hover:tw-opacity-75;
    @apply focus:tw-ring;
    @apply focus:tw-ring-blue-200;
    @apply active:tw-opacity-[0.85];
  }
}
