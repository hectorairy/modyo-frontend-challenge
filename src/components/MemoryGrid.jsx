import { useEffect, useState } from "react";
import { MemoryCard } from "./MemoryCard";
import { IMG_API } from "../utils/constants";

export const MemoryGrid = () => {
  const [cards, setCards] = useState([]);

  useEffect(() => {
    fetch(IMG_API)
      .then((res) => res.json())
      .then(({ entries }) => {
        const items = entries.map((item) => ({
          id: item.meta.uuid,
          name: item.meta.name,
          url_image: item.fields.image.url,
          stat: "",
        }));
        const newItems = [...items, ...items].sort(() => Math.random() - 0.5);
        setCards(newItems);
      })
      .catch((e) => console.error(e));
  }, []);

  return (
    <>
      <div className="memory-game__grid">
        {cards.map((card, index) => (
          <MemoryCard key={index} card={card} />
        ))}
      </div>
    </>
  );
};
