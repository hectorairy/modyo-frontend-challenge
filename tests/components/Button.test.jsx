import { render, screen } from "@testing-library/react";
import { Button } from "../../src/components/Button";

describe("<Button /> test", () => {
  test("should match with snapshot ", () => {
    const { container } = render(
      <Button handleClick={() => {}}>Hello World!</Button>
    );

    expect(container).toMatchSnapshot();
  });
});
