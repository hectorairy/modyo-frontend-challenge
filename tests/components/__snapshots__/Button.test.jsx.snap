// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> test should match with snapshot  1`] = `
<div>
  <button
    class="custom-button"
  >
    Hello World!
  </button>
</div>
`;
